libavfilter/x86/vf_convolution_init.c

```diff
@@ -44,3 +49,17 @@ av_cold void ff_convolution_init_x86(ConvolutionContext *s)
     }
 #endif
 }
+
+av_cold void ff_sobel_init_x86(ConvolutionContext *s)
+{
+#if ARCH_X86_64
+    int i;
+    int cpu_flags = av_get_cpu_flags();
+    for (i = 0; i < 4; i++) {
```

Comment:
does s always have 4 planes ? You'd be better to change the condition to 'i < s->nb_planes'

Reply:
Community prefers to declare loop indices in the loop statement itself

for (int i = 0; ... )

Reply:
Solved in the new commit